### ISSUE TITLE
[Bug]: Split Error

### ISSUE THREAD

**Issue by psvajaz** [rikyoz/bit7z#161]
### bit7z version

4.0.x RC

### Compilation options

BIT7Z_AUTO_FORMAT, BIT7Z_USE_NATIVE_STRING

### 7-zip version

v22.01

### 7-zip shared library used

7z.dll / 7z.so

### Compilers

MSVC

### Compiler versions

MSVC2017

### Architecture

x86_64

### Operating system

Windows

### Operating system versions

Windows7 or last

### Bug description

If there is an exception in compressing a volume or compressing a package, there will be an exception during the preview process, as the code 'const noexception' has been added to the exception code, resulting in external failure to capture exception information properly
![image](https://github.com/rikyoz/bit7z/assets/88539/3b4a8ed1-bef4-4114-80c2-fd9e7b22596b)

### Code of Conduct

- [X] By submitting this issue, I agree to follow bit7z's [Code of Conduct](https://github.com/rikyoz/bit7z/blob/master/CODE_OF_CONDUCT.md)

**Comment by rikyoz** [comment]
Hi!
This is strange! Could you please provide the full stack trace of the error?

**Comment by psvajaz** [comment]
This is caused by an abnormal environment. When the volume compression (any compression tool) is set to ZIP, the compression program unexpectedly exits, and the compressed files retained on the hard drive may experience this situation during preview execution. 
Due to the compressed files provided by users, there may be various situations. Should error information be provided when unrecognized. Otherwise, there will be a program crash

![image](https://github.com/rikyoz/bit7z/assets/88539/5350c0cd-9733-4716-a503-193b40f87360)

**Comment by rikyoz** [comment]
Thanks for the details!

I am having some difficulties in trying to replicate the problem.
Just to be sure that I correctly understand it:

+ You create a multi-volume/split ZIP archive using any compression tool (e.g., 7z.exe).
+ For some reason, the tool unexpectedly exits before finishing the compression operation, leaving either some missing volumes or possibly some damaged/incomplete volume files.
+ When you try to read the (incomplete/damaged) multi-volume archive using bit7z, you have that unexpected exception.

Did I understand it correctly?

This is strange. In all my attempts at replicating the issue, bit7z correctly reported some data errors while previewing the archive. But I'm probably missing some detail.

The stack trace you provided is even stranger. The `operator ==` of `BitInFormat` is `const noexcept` simply because... It should not throw any exception in typical situations and doesn't modify the member variables.
The fact that `this` is `nullptr` is even stranger since the operator should have been called on two valid references, not on two pointers.
I'll need to investigate the issue even further.

It would be handy if you could send me the code snippet you're using and, possibly, the multi-volume archive files giving you the issue. If you can't send the latter, don't worry, I understand!

Also, since I recently made some small changes to the format detection code, you might want to test out bit7z from the [v4.0.0 branch](https://github.com/rikyoz/bit7z/tree/release/v4.0.0). Maybe it doesn't have the same issue.

I really appreciate any help you can provide. 🙏

**Comment by psvajaz** [comment]
During the testing process, it was found that if the number of volumes exceeds 8188, an error will occur and the compressed volumes will remain on the disk. When using the program to preview and decompress, this exception will occur

**Comment by rikyoz** [comment]
> if the number of volumes exceeds 8188, an error will occur.

Yeah, I will work on a fix for this limit, but not in the stable version v4.0.

> When using the program to preview and decompress, this exception will occur

This is strange, because an exception should occur, but it should be a "normal" BitException due to a data error. How many files are compressed in the multi-volume archive? What kind of files are they? And how big are they?
